### DISCUSSION TITLE
Power Limits not working on a R9 7945HX (2023 Zephyrus Duo16 CPU)

### DISCUSSION THREAD

**Discussion by Provinomico** [seerge/g-helper#708] [General]
Hey Serge,

Thanks for your amazing and constant work!

A week ago I tested the latest version and was excited to discover the new dynamic boost settings for the GPU, which didn't work for me at first but did after I updated to the latest build.

Today I ran a test to check the CPU power limits settings, and it seems that are not applying correctly. On Cinebench multi core it topped at 85w, all core frequency went up to stable 4.45Ghz.

OCCT Package Power
![image](https://github.com/seerge/g-helper/assets/137456477/2a4a8b61-402b-4909-9b2f-88b1540019b7)

G-Helper CPU Power Limits
![image](https://github.com/seerge/g-helper/assets/137456477/76a38917-a589-4221-93c8-71de6c715608)

On a AC maxed out CPU profile, wattage peaked at 129w, and all core frequency reached 5.55-5.6Ghz which gradually went down to a stable 4.8Ghz by the end of the run

I'm on 0.91.0.0 btw. If you want me to provide traces or run test builds to try stuff just drop me a message.

**Comment by seerge** [comment]
@Provinomico hello. Can you post your log and config from %appdata%/ghelper?

**Comment by Provinomico** [comment]
Sure, it's the 'full blast' profile. I tried both at 130w which is the CPU/Bios hard limit, and maxing out to the 150w that allow the sliders, with same results.

[config.zip](https://github.com/seerge/g-helper/files/11926560/config.zip)

**Reply by seerge** [reply]
@Provinomico if you set same limits just in Turbo mode (not custom one), does it behave same? 

If so, try to quit the app, add following line to the top of config.json (incl comma in the end), save it
```
"manual_mode" : 1,
```

And run app again (and try limits again / post logs here) :)

**Reply by Provinomico** [reply]
OK, played a bit with that setting, same result. Things to note:

The Turbo profile gets up to (and kind of maintains) 92w peak watts, while the custom one tops at 88w, no matter the sliders being at 130/150w. I've run the test several times to account for margin of error, all the runs result in the same readings. If it were the other way around an explanation would be the custom profile having 100% fan speed at the whole range of temps, as heat plays a role on the CPU controlling itself. But somehow, with worse temps the turbo profile nets +5w of CPU power.

The sliders do work in the lower ranges, I've used it to set the 'Balanced Quiet' profile in which I limit CPU wattage below the 'Balanced's default.

Along with the 'manual_mode' I also added the 'max_gpu_memory' option you told me about to be able to set higher memory clock. I saw the slider range change once I restarted the app so I guess config was read ok.

In AC I remember there were 3 sliders in regard to CPU limits, I'll try installing it again to check what they were. Also to check if AC leave some kind of log that might help in tracking which endpoint it uses to set values.

Here's the config/logs after the last changes.

[GHelper.zip](https://github.com/seerge/g-helper/files/11926650/GHelper.zip)

**Reply by seerge** [reply]
@Provinomico oke, thanks for testing. Since sliders work in lower side - it looks like all endpoints are correct :)

AC indeed has 3 sliders STAMP / sPPT / fPPT, in g-helper i just merged first 2 into 1 (in fact it sets 3 values)

I'm rather keen to think that there is some hard limit for each "mode" in bios. So try AC along with g-helper and see if it makes difference (may be it has some "hidden" mode with unlimited power that it uses for "manual" in AC?)

But before that, i see that there is aggressive undervolting settings, try to set them to 0, and for clean experiment even reboot laptop after to see if that makes a difference (even tho it shouldn't)

**Reply by Provinomico** [reply]
I'll continue testing with AC and report back.

If the mode plays a role on the unlocking of higher wattages it might be just that. I think in AC you would go to the manual tab, and create custom profiles. That's what you thought it was when you asked me to add that 'manual_mode' setting? is there a way to monitor ACPI call to check what AC uses when setting up/activating profiles?

About the aggressive undervolting, I might have won the silicon lottery with this one, at -30 Cinebench runs the 30min stability test just fine. On a single pass I could go as high as -50  getting a +37k score which is bonkers. Highest I was able to run the stability test was -40, but I don't want to push it that far.

I got -30 dialed on the bios since the first week of purchase, haven't had any stability issues so far.

**Comment by envyska** [comment]
@Provinomico  Try to uninstall Asus controll interface 3.1.1.14.0 from device manager then reinstall 3.1.1.13.0. It worked for me on all kind of g helper not working issues including power limits/ custom fan curve issue.

Device manger -> System device -> delete asus control interface V3 + driver -> reboot -> reinstall 3.1.1.13.0 -> reboot -> good

**Reply by Provinomico** [reply]
@envyska Superb!!! will give that a try before installing AC again, fingers crossed!!

**Comment by Provinomico** [comment]
I'm updating the thread with some info for future reference. So @envyska , going back to the 3.1.1.13.0 of Asus control Interface didn't resolve the issue. Went back and forth a few times, no change.

In the end I completed a fresh install of AC, now it's working again and the CPU is able to reach the max tdp of 130w both with AC and G-Helper. I'll now slowly disable all the shitload of services, checking if everything continues to work properly one at a time. Maybe I shouldn't have uninstalled AC, just leave the installation be, but with everything disabled.

@seerge one thing I discovered, after the testing I did las weekend, the GPU tdp was no longer linked to each profile and it just went all the way up to 150w, always, no matter the profile, also, ignoring the boost value (should reach ~175w accounting for the boost). Checking what I had changed in the last day, I confirmed it was that  "manual_mode" : 1 option I added to the config to see if I could fix the issue with the CPU max TDP. Whatever that option does, it makes the GPU go all the way up minus the boost. So, question, is there a way to set GPU tdp levels? or are those hardcoded on the bios?

**Reply by seerge** [reply]
If you had power limits enabled in each mode (silent / balanced / turbo) it will indeed turn on "manual" mode instead (so effectively all 3 modes will have same thing as base), so no suprise GPU limits are same.

Since manual mode flag does nothing for you anyway, you can safely remove it from config :) as it's not your case anyway

As for general case when power limits didn't work before, and now do - i dont' know what's the reason honestly :) I could be that it was some leftower from AC messing with settings, and now when you reinstalled everything back that "leftower" is under control at least :))

Try to uninstall AC (if you want ofc), and see if it will still be ok
